src/main/java/uk/gov/hmcts/reform/ethos/ecm/consumer/tasks/WorkQueueTask.java

```diff
@@ -0,0 +1,180 @@
+            }
+            return;
+        }
+
+        log.info("RECEIVED 'Create Updates' ------>  message with ID {}", message);
+
+        try {
+            if (message.getDataModelParent() instanceof TransferToEcmDataModel) {
+                transferToEcmService.transferToEcm(message);
+            } else {
+                sendUpdateCaseMessages(message);
+            }
+            setComplete(conn, work.getId());
+        } catch (Exception e) {
+            setErrored(conn, work.getId());
+            log.error("Error processing work item", e);
+        }
+    }
+
+    private void sendUpdateCaseMessages(CreateUpdatesMsg createUpdatesMsg)
+            throws SQLException {
+        if (createUpdatesMsg.getEthosCaseRefCollection() == null) {
+            return;
+        }
+
+        try (Connection conn = etcosDataSource.getConnection()) {
+            for (String ethosCaseReference : createUpdatesMsg.getEthosCaseRefCollection()) {
+                UpdateCaseMsg updateCaseMsg = UpdateCaseMsg.builder()
+                        .msgId(UUID.randomUUID().toString())
+                        .multipleRef(createUpdatesMsg.getMultipleRef())
+                        .ethosCaseReference(ethosCaseReference)
+                        .totalCases(createUpdatesMsg.getTotalCases())
+                        .multipleReferenceLinkMarkUp(createUpdatesMsg.getMultipleReferenceLinkMarkUp())
+                        .jurisdiction(createUpdatesMsg.getJurisdiction())
+                        .caseTypeId(createUpdatesMsg.getCaseTypeId())
+                        .username(createUpdatesMsg.getUsername())
+                        .confirmation(createUpdatesMsg.getConfirmation())
+                        .dataModelParent(createUpdatesMsg.getDataModelParent())
+                        .build();
+
+                sendMessage(conn, updateCaseMsg);
+            }
+        } catch (SQLException ex) {
+            log.error(ex.getMessage());
+            throw ex;
+        }
+    }
+
+    private void sendMessage(Connection conn, UpdateCaseMsg msg) {
+        try (CallableStatement addWork = conn.prepareCall("{ call add_work(?, ?) }")) {
+            addWork.setString(1, msg.getMultipleRef());
+
+            PGobject jsonObject = new PGobject();
+            jsonObject.setType("json");
+            jsonObject.setValue(new ObjectMapper().writeValueAsString(msg));
+            addWork.setObject(2, jsonObject);
+
+            addWork.execute();
+        } catch (SQLException | JsonProcessingException ex) {
+            log.error("Failed to add work", ex);
+        }
+    }
+
+    private void setComplete(Connection conn, int id) {
+        try (CallableStatement call = conn.prepareCall("{ call complete_work(?) }")) {
+            Array sqlArray = conn.createArrayOf("int4", new Integer[] { id });
+            call.setArray(1, sqlArray);
+            call.execute();
+        } catch (SQLException ex) {
+            log.error("Failed marking work as errored", ex);
```

Comment:
Should this be "Failed marking work as complete"?

Reply:
Good spot, it should be